client_apis/python/src/tests/integration/watchlist_test.py

```diff
@@ -0,0 +1,420 @@
+#
+# CARBON BLACK API TESTS - watchlist
+# Copyright, Bit9, Inc 2015
+#
+
+""" These tests require CarbonBlack Enterprise Server to be installed.
+    You can run this script by passing the server URL and user API token
+    as parameters.
+"""
+
+import unittest
+import sys
+import os
+import requests
+from datetime import datetime
+
+if __name__ == '__main__':
+    sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../")))
+
+from cbapi.cbapi import CbApi
+
+cb = None
+name_prefix = "Test Watchlist"
+
+
+class CbApiWatchlistTest(unittest.TestCase):
+    def test_get_watchlists(self):
+        watchlists = cb.watchlist()
+        self.assertIsNotNone(watchlists)
+        return
+
+    def test_add_watchlist(self):
+        old_watchlists = cb.watchlist()
+        self.assertIsNotNone(old_watchlists)
+        add_result = cb.watchlist_add(
+            type="modules",
+            name="%s %s" % (name_prefix, datetime.now().isoformat(' ')),
+            search_query="q=is_executable_image%3Afalse&cb.urlver=1&sort=server_added_timestamp%20desc",
+            readonly=False
+        )
+        self.assertGreater(add_result["id"], 0)
+
+        new_watchlists = cb.watchlist()
+        self.assertEqual(len(new_watchlists), len(old_watchlists) + 1)
+        return
+
+    def test_add_invalid_watchlist(self):
+        old_watchlists = cb.watchlist()
+        self.assertIsNotNone(old_watchlists)
+        with self.assertRaises(Exception):
+            cb.watchlist_add(
+                type="bad_type",
+                name="%s %s" % (name_prefix, datetime.now().isoformat(' ')),
+                search_query="q=is_executable_image%3Afalse&cb.urlver=1&sort=server_added_timestamp%20desc",
+                readonly=False
+            )
+
+        new_watchlists = cb.watchlist()
+        self.assertEqual(len(new_watchlists), len(old_watchlists))
+        return
+
+    def test_get_watchlist(self):
+        watchlists = cb.watchlist()
+        self.assertIsNotNone(watchlists)
+        for watchlist in watchlists:
+            result = cb.watchlist(id=watchlist["id"])
+            self.assertIsNotNone(result)
+        return
+
+    def test_get_invalid_watchlist(self):
+        with self.assertRaises(requests.HTTPError) as err:
+            cb.watchlist(id="bad_id")
+        self.assertEqual(err.exception.response.status_code, 400)
+
+        with self.assertRaises(requests.HTTPError) as err:
+            cb.watchlist(id=-1)
+        self.assertEqual(err.exception.response.status_code, 404)
+        return
+
+    def test_modify_watchlist(self):
+        add_result = cb.watchlist_add(
```

Comment:
You may want to consider using a helper method for this block since it's used several times for other methods.